lib/find_pkg.ts

```diff
@@ -14,21 +14,35 @@ function findPkg(cwd: string, args: findPkgArgs = {}) {
 }
 
 function checkPkg(path: string) {
-  const pkgPath = join(path, 'package.json');
-
-  return readFile(pkgPath).then(content => {
-    const json = JSON.parse(content as string);
-    if (typeof json.hexo === 'object') return path;
-  }).catch(err => {
-    if (err && err.code === 'ENOENT') {
-      const parent = dirname(path);
-
-      if (parent === path) return;
-      return checkPkg(parent);
-    }
-
-    throw err;
-  });
+  // prefers package.json over package.yaml
+  const pkgJsonPath = join(path, 'package.json');
+  const pkgYamlPath = join(path, 'package.yaml');
+
+  return readFile(pkgJsonPath)
+    // if package.json exists, read it
+    .then(content => JSON.parse(content))
+    .catch(err => {
+      if (err.code === 'ENOENT') {
```

Comment:
Let's just use `fs.existsSync` before reading the file. It is significantly faster.